CFE-3629 - just poking around

btriller opened the pull request:
Set CFLAGS before AC_PROG_CC as it might add -g/-O options if CFLAGS is unset.
Don't add CFLAGS to CORE_FLAGS because I don't know, but maybe CFLAGS is populated otherwise.

btriller committed:
CFE-3629 - just poking around

Set CFLAGS before AC_PROG_CC as it might add -g/-O options if CFLAGS is
unset.
Don't add CFLAGS to CORE_FLAGS because I don't know, but maybe CFLAGS is
populated otherwise.

olehermanse commented:
@btriller Why did you close this? IIRC, the change looked reasonable, but more work was needed to fix the failing tests(?)

btriller commented:
> @btriller Why did you close this? IIRC, the change looked reasonable, but more work was needed to fix the failing tests(?)

I looked further into it on macOS (with clang). With `-Wall -Werror`, there were many errors about redefinition that I wasn't sure I'm able to resolve it. IIRC on Linux with gcc and, for starters, more ifdefs in libcfecompat, it looked better.

btriller committed:
Fix redefinition errors

btriller committed:
Fix CFLAGS

btriller committed:
Fix pointer-sign warnings

btriller committed:
Fix format truncation warning

btriller committed:
Remove more duplicate flags

btriller committed:
Keep autogen/configure CFLAGS

In case CFLAGS were given to configure, keep them for later stages.
Check for clang compiler so strchrnul() can be used for macOS >= 15.4.
Replace some obsolete autoconf macros.
Make -Werror works in configure.

olehermanse commented:
@btriller Thanks for working on this. We're a bit busy with getting the releases out, but I'll get back to reviewing the PR shortly after that :)
